File: src/services/auth.service.js
import bcrypt from "bcryptjs";
import jwt from "jsonwebtoken";
import { env } from "../config/env.js";
import { Config } from "../models/Config.js";

async function getConfig() {
  let config = await Config.findOne();
  if (!config) {
    const passwordHash = await bcrypt.hash(env.defaultPassword, 10);
    config = await Config.create({ passwordHash });
    console.log(`Contraseña por defecto inicializada: "${env.defaultPassword}"`);
  }
  return config;
}

export const authService = {
  async login({ password }) {
    const config = await getConfig();
    const valid = await bcrypt.compare(password, config.passwordHash);
    if (!valid) {
      const err = new Error("Contraseña incorrecta");
      err.statusCode = 401;
      throw err;
    }
    const token = jwt.sign({ sub: "seekpal" }, env.jwtSecret, { expiresIn: env.jwtExpiresIn });
    return { accessToken: token, tokenType: "Bearer", expiresIn: env.jwtExpiresIn };
  },

  async changePassword({ currentPassword, newPassword }) {
    const config = await getConfig();
    const valid = await bcrypt.compare(currentPassword, config.passwordHash);
    if (!valid) {
      const err = new Error("Contraseña actual incorrecta");
      err.statusCode = 401;
      throw err;
    }
    config.passwordHash = await bcrypt.hash(newPassword, 10);
    await config.save();
  },
};

File: backend/app/services/auth_service.py
import logging

from app.core.config import settings
from app.core.responses import APIError
from app.core.security import (
    create_access_token,
    hash_password,
    verify_password,
)
from app.models.config import Config

logger = logging.getLogger("seekpal.auth")


async def get_or_create_config() -> Config:
    config = await Config.find_one()
    if config is None:
        config = Config(passwordHash=hash_password(settings.default_password))
        await config.insert()
        logger.info('Contraseña por defecto inicializada: "%s"', settings.default_password)
    return config


async def login(password: str) -> dict:
    config = await get_or_create_config()
    if not verify_password(password, config.passwordHash):
        raise APIError("Contraseña incorrecta", status_code=401)
    token = create_access_token()
    return {
        "accessToken": token,
        "tokenType": "Bearer",
        "expiresIn": f"{settings.jwt_expires_minutes}m",
    }


async def change_password(current: str, new: str) -> None:
    config = await get_or_create_config()
    if not verify_password(current, config.passwordHash):
        raise APIError("Contraseña actual incorrecta", status_code=401)
    config.passwordHash = hash_password(new)
    await config.save()
